### PR TITLE
Task-43069/Task-43070 : Add the possibility to suspend temporarily documents sharing and documents download using transfert rules page.

### PR DESCRIPTION
**Pull request by ahmadmido**
This PR will add the possibility to transfers rules administrator to suspend temporarily : 
- The share documents actions.
-  The download documents actions.

### CODE REVIEW

File: apps/portlet-transferrules/src/main/webapp/vue-apps/transferRules/components/TransferRulesApp.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
   <v-app id="transferRules">
-    <v-card class="ma-4 border-radius" flat>
+    <v-card class="ma-4 border-radius gggggBtn" flat>
```
Comment (plamarque) [inline apps/portlet-transferrules/src/main/webapp/vue-apps/transferRules/components/TransferRulesApp.vue]: typo ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 2]
```diff
@@ -0,0 +1,167 @@
+@import "../../../variables.less";
+
+#transferRules {
+
+  .gggggBtn {
```
Comment (plamarque) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less]: why so many gs?

File: core/services/src/main/java/org/exoplatform/services/rest/transferRules/TransferRulesRestService.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                                                      "exo:sharedDocumentStatus");
+    boolean isSharedDocumentActivated = settingValue != null && !settingValue.getValue().toString().isEmpty() ? Boolean.valueOf(settingValue.getValue().toString()) : true;
+    return Response.ok().entity("{\"isSharedDocumentActivated\":\"" + isSharedDocumentActivated + "\"}").build();
+  }
+
+  @PUT
+  @Path("/saveSharedDocumentStatus")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "Updates the share document status",
```
Comment (plamarque) [inline core/services/src/main/java/org/exoplatform/services/rest/transferRules/TransferRulesRestService.java]: shouldn't  this be permitted  to admins only?

**Review by rdenarie — CHANGES_REQUESTED** [review]
When we create an action in ECMs, we can define filters like here : https://github.com/exoplatform/ecms/blob/d81b5b525bfaf5f6d3383b02e82e4c4700adb24b/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/EditDocumentActionComponent.java#L87
My idea was to create a new Filter for TransferRules, and add it in ShareDocumentsComponent
So, please can you modify the PR to use the filter system ?

File: apps/portlet-transferrules/src/main/webapp/vue-apps/transferRules/components/TransferRulesApp.vue [hunk 4]
```diff
@@ -32,14 +80,34 @@
...
+import * as transferRulesService from '../transferRulesService.js';
+
 export default {
   data () {
     return {
-      active: true,
+      sharedDocumentStatus: true,
+      downloadDocumentStatus: true,
+      uploadDocumentStatus: true,
+      TransferRulesStatusModel: null,
```
Comment (rdenarie) [inline apps/portlet-transferrules/src/main/webapp/vue-apps/transferRules/components/TransferRulesApp.vue]:
```suggestion
      transferRulesStatusModel: null,
```

**Review by rdenarie — CHANGES_REQUESTED** [review]
- check rtl css
- check color variables in css
- UP for view.gtmpl

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 5]
```diff
@@ -0,0 +1,163 @@
+@import "../../../variables.less";
+
+#transferRules {
+
+  /* switch */
+  label.uiSwitchBtn {
+      height: 30px;
+  }
+  .uiSwitchBtn span.deactivated {
+      text-align: left;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less]: Should it be rtlized ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+      transition: .4s;
+  }
+
+  .slider:before {
+      position: absolute;
+      z-index: 2;
+      content: "";
+      height: 14px;
+      width: 14px;
+      left: 5px;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less:45]: Should it be rtlized ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+      bottom: 3px;
+      background-color: @darkGrey;
+      -webkit-box-shadow: 0 2px 5px rgba(0, 0, 0, 0.22);
+      box-shadow: 0 2px 5px rgba(0, 0, 0, 0.22);
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less:49]: Should it be rtlized ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+      transition: all 0.4s ease-in-out;
+  }
+
+  input:not(:checked)+.slider > .absolute-activate {
+    z-index: -1;
+    position: absolute;
+    -webkit-transform: translateX(-190px);
+    -ms-transform: translateX(-190px);
+    transform: translateX(-190px);
+    left: 3px;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less:60]: Should it be rtlized ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+    transition: all 0.4s ease-in-out;
+  }
+
+  input:checked+.slider> .absolute-activate {
+    -webkit-transform: translateX(0px);
+    -ms-transform: translateX(0px);
+    transform: translateX(0px);
+    z-index: 15;
+    position: absolute;
+    left: 3px;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less:76]: Should it be rtlized ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+    font-size: 12px;
+    width: calc(100% - 50px);
+    line-height: 22px;
+    cursor: pointer;
+    transition: all 0.4s ease-in-out;
+  }
+
+  .slider:after {
+    position: absolute;
+    left: -20px;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less:88]: Should it be rtlized ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    z-index: 1;
+    content: "";
+    font-size: 13px;
+    text-align: left !important;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less]: Should it be rtlized ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+    -ms-transform: translateX(-190px);
+    transform: translateX(-190px);
+    transition: all 0.4s ease-in-out;
+  }
+
+  input:checked+.slider:after {
+    -webkit-transform: translateX(0px);
+    -ms-transform: translateX(0px);
+    transform: translateX(0px);
+    padding-left: 25px;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less:127]: Should it be rtlized ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+    border-radius: 100px;
+  }
+
+  .slider.round:before {
+    border-radius: 50%;
+  }
+
+  .absolute-deactivated {
+    position: absolute;
+    left: 8px;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less]: Should it be rtlized ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you also try to reduce css part in transfer-rules.less, by using vuejs css helper ?
https://vuetifyjs.com/en/styles/css-reset/

You can remove at least all "position", left, right, margin/padding,border radius

File: core/services/src/main/java/org/exoplatform/services/rest/transferRules/TransferRulesRestService.java [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+
+  private SettingService settingService;
+
+  public TransferRulesRestService(SettingService settingService) {
+    this.settingService = settingService;
+  }
+
+  @GET
+  @Path("/getTransfertRulesDocumentStatus")
+  @RolesAllowed("users")
```
Comment (rdenarie) [inline core/services/src/main/java/org/exoplatform/services/rest/transferRules/TransferRulesRestService.java:27]: should it be administrators ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+  }
+
+  /* switch */
+  label.uiSwitchBtn {
+      height: 30px;
+  }
+
+  .switch {
+      position: relative;
+      display: inline-block;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less]: can you use the class vue "d-inline-block"

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+      width: 53px;
+      height: 32px;
+      bottom: 7px;
+  }
+
+  .switch input {
+      display: none;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less]: can you use the class d-none

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+    transition: all 0.4s ease-in-out;
+  }
+
+  .slider:after {
+    position: absolute;
+    left: -20px;
+    z-index: 1;
+    content: "";
+    font-size: 13px;
+    text-align: left !important;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less]: can you use class text-left

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+    -ms-transform: translateX(-190px);
+    transform: translateX(-190px);
+    transition: all 0.4s ease-in-out;
+  }
+
+  input:checked+.slider:after {
+    -webkit-transform: translateX(0px);
+    -ms-transform: translateX(0px);
+    transform: translateX(0px);
+    padding-left: 25px;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less:127]: can you use class pl-25 if it is always left or ps-25 if it should change with RTL ?

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less [hunk 19]
```diff
@@ -0,0 +1,163 @@
...
+  }
+
+  .slider:before {
+    border-radius: 50%;
+  }
+
+  .absolute-deactivated {
+    position: absolute;
+    color: @darkGrey;
+    text-align: right ;
```
Comment (rdenarie) [inline apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less]: can you use class text-right

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less (hunk 18 above)
Comment (ahmadmido) [reply apps/resources-wcm/src/main/webapp/skin/less/ecms/portlets/transferRules/transfer-rules.less:127]: It's affect the button style, i can't use vuetify class here, it depend to bouton animation (input:checked+.slider:before and input:checked+.slider:after)

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**